src/routers/content.js

```diff
@@ -0,0 +1,240 @@
+// Handler to assemble a specific piece of static content.
+
+var
+  fs = require('fs'),
+  path = require('path'),
+  request = require('request'),
+  url = require('url'),
+  urljoin = require('url-join'),
+  async = require('async'),
+  handlebars = require('handlebars'),
+  _ = require('lodash'),
+  config = require('../config'),
+  logger = require('../server/logging').logger,
+  TemplateService = require('../services/template'),
+  TemplateRoutingService = require('../services/template-routing'),
+  ContentRoutingService = require('../services/content/routing'),
+  ContentFilterService = require('../services/content/filter'),
+  UrlService = require('../services/url'),
+  HttpErrorHelper = require('../helpers/http-error');
+
+// Create an Error object with the provided message and a custom attribute that remembers the
+// (presumably non-200) status code of the associated HTTP response.
+function response_error(res, message) {
+var err = new Error(message);
+err.statusCode = res.statusCode;
+
+if (err.statusCode === 404) {
+  err.statusMessage = "Required resource not found";
+} else {
+  err.statusMessage = "Upstream server error";
+}
+
+return err;
```

Comment:
Yeah, I'm trying to deprecate `response_error` and `content_error` to use `HttpErrorHelper` if possible. I think the tab-squash came from me removing them, leaving a hanging reference, panicking, then hitting Ctrl-Z.